dev/io.openliberty.org.jboss.resteasy.cdi.ee10/src/org/jboss/resteasy/cdi/ResteasyCdiExtension.java

```diff
@@ -0,0 +1,346 @@
+package org.jboss.resteasy.cdi;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import jakarta.decorator.Decorator;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.Any;
+import jakarta.enterprise.inject.Default;
+import jakarta.enterprise.inject.spi.AfterBeanDiscovery;
+import jakarta.enterprise.inject.spi.AfterTypeDiscovery;
+import jakarta.enterprise.inject.spi.AnnotatedType;
+import jakarta.enterprise.inject.spi.Bean;
+import jakarta.enterprise.inject.spi.BeanManager;
+import jakarta.enterprise.inject.spi.BeforeBeanDiscovery;
+import jakarta.enterprise.inject.spi.Extension;
+import jakarta.enterprise.inject.spi.InjectionTarget;
+import jakarta.enterprise.inject.spi.ProcessAnnotatedType;
+import jakarta.enterprise.inject.spi.ProcessBean;
+import jakarta.enterprise.inject.spi.ProcessInjectionTarget;
+import jakarta.enterprise.inject.spi.ProcessSessionBean;
+import jakarta.enterprise.inject.spi.WithAnnotations;
+import jakarta.enterprise.util.AnnotationLiteral;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.cdi.i18n.LogMessages;
+import org.jboss.resteasy.cdi.i18n.Messages;
+import org.jboss.resteasy.plugins.providers.RegisterBuiltin;
+
+/**
+ * This Extension handles default scopes for discovered JAX-RS components. It
+ * also observes ProcessInjectionTarget event and wraps InjectionTargets
+ * representing JAX-RS components within JaxrsInjectionTarget. Furthermore, it
+ * builds the sessionBeanInterface map which maps Session Bean classes to a
+ * local interface. This map is used in CdiInjectorFactory during lookup of
+ * Sesion Bean JAX-RS components.
+ *
+ * @author Jozef Hartinger
+ *
+ */
+public class ResteasyCdiExtension implements Extension {
+   private static boolean active;
+   private static final String JAKARTA_EJB_STATELESS = "jakarta.ejb.Stateless";
+   private static final String JAKARTA_EJB_SINGLETON = "jakarta.ejb.Singleton";
+
+   // Scope literals
+   private static final Annotation requestScopedLiteral = new AnnotationLiteral<RequestScoped>() {
+      private static final long serialVersionUID = 3381824686081435817L;
+   };
+   public static final Annotation applicationScopedLiteral = new AnnotationLiteral<ApplicationScoped>() { //Liberty change
```

Comment:
maybe add a comment why this needs to be changed. Protected might be better. We could also submit a PR to Resteasy to make this `protected` with the justification that we (and others) need it to use this class as an extension point.

Reply:
Agreed

Reply:
If we do a PR for Resteasy, we could also make those duplicated private utility methods protected as well.